src/auto-router.ts

```diff
@@ -526,20 +541,20 @@ export class AutoRouter {
     const recipient = await this.indexRouter.signer.getAddress();
     const chainId = await this.indexRouter.signer.getChainId();
 
-    const isBuyingNative = buyToken.toLowerCase() === NATIVE.toLowerCase();
-    const routerBuyToken = isBuyingNative ? await this.indexRouter.contract.WETH() : buyToken;
+    const isNativeBuy = isNative(buyToken as Address);
+    const routerBuyToken = isNativeBuy ? await this.indexRouter.contract.WETH() : buyToken;
 
     const amounts = await this.indexRouter.burnAmount(indexToken, sellAmount);
 
     const quotes = await Promise.all(
       amounts.map(async ({ amount, asset }) => {
-        if (!amount || amount.isZero() || asset.toLowerCase() === routerBuyToken.toLowerCase()) {
+        if (isAddressEqual(asset as Address, routerBuyToken as Address) || !amount || amount.isZero()) {
```

Comment:
`!amount`
I am not sure that amount can be nullable/undefined, can you check it please?